src/script/descriptor.cpp

```diff
@@ -1546,8 +1556,8 @@ std::unique_ptr<DescriptorImpl> ParseScript(uint32_t& key_exp_index, Span<const
         }
         ++key_exp_index;
         return std::make_unique<PKHDescriptor>(std::move(pubkey));
-    } else if (Func("pkh", expr)) {
-        error = "Can only have pkh at top level, in sh(), or in wsh()";
+    } else if (ctx != ParseScriptContext::P2TR && Func("pkh", expr)) {
+        error = "Can only have pkh at top level, in sh(), wsh(), or in tr()";
```

Comment:
In ffdca88818f77fda06c6331df5b4bb408ab1833b "descriptor: parse Miniscript expressions within Taproot descriptors"

This doesn't actually allow `pkh()` to be parsed by the descriptor parsing code, but I assume that's because the point is to let the miniscript parser deal with it. In that case, a comment here would be helpful.

Reply:
Added a comment to this effect.